atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1976,6 +1979,18 @@ public void useTable(TableReference tableRef, ConstraintCheckable table) {
         constraintsByTableName.put(tableRef, table);
     }
 
+    /** the similarly-named-and-intentioned useTable method is only called on writes,
+     *  this one is more comprehensive and covers read paths as well
+     * (necessary because we wish to get the sweep strategies of tables in read-only transactions)
+     */
+    private void markTableAsInvolvedInThisTransaction(TableReference tableRef) {
+        involvedTables.add(tableRef);
+    }
+
+    private boolean validationNecessaryForInvolvedTables() {
+        return !Sets.filter(involvedTables, tableRef -> isValidationNecessary(tableRef)).isEmpty();
```

Comment:
nit: might be clearer as:

`return involvedTables.stream().anyMatch(this::isValidationNecessary)`